src/tiled/commanddialog.cpp

```diff
@@ -63,6 +69,19 @@ void CommandDialog::closeEvent(QCloseEvent *event)
     CommandManager::instance()->updateActions();
 }
 
+void CommandDialog::setShortcut(const QKeySequence &keySequence)
+{
+    mUi->treeView->model()->setShortcut(mUi->treeView->currentIndex(), keySequence);
+}
+
+void CommandDialog::updateKeySequenceEdit(const QModelIndex &current, const QModelIndex &)
+{
+    if (current.row() < mUi->treeView->model()->rowCount(current))
```

Comment:
The segmentation fault would be because you're calling it when there is no valid current index, but clearing the shortcut changes it and then you're calling CommandDataModel::setShortcut with an invalid index, which then tries to set the shortcut of the command at -1.